### PR TITLE
feat(auth): add options to resendSignUpCode

### PR DESCRIPTION
**Pull request by Jordan-Nelson**
*Issue #, if available:* n/a

*Description of changes:*
- add options for resendSignUpCode to enable the use of client metadata

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/amplify_auth_cognito/android/src/test/kotlin/com/amazonaws/amplify/amplify_auth_cognito/AmplifyAuthCognitoPluginTest.kt [hunk 1]
```diff
@@ -21,18 +21,16 @@ import com.amazonaws.amplify.amplify_core.exception.InvalidRequestException
...
 import com.amplifyframework.auth.cognito.AWSCognitoAuthSession
 import com.amplifyframework.auth.cognito.AWSCognitoUserPoolTokens
 import com.amplifyframework.auth.cognito.options.AWSCognitoAuthConfirmSignInOptions
 import com.amplifyframework.auth.cognito.options.AWSCognitoAuthConfirmSignUpOptions
+import com.amplifyframework.auth.cognito.options.AWSCognitoAuthResendSignUpCodeOptions
 import com.amplifyframework.auth.cognito.options.AWSCognitoAuthSignInOptions
-import com.amplifyframework.auth.options.AuthConfirmSignInOptions
-import com.amplifyframework.auth.options.AuthConfirmSignUpOptions
-import com.amplifyframework.auth.options.AuthSignInOptions
+import com.amplifyframework.auth.options.*
```
Comment (haverchuck) [inline packages/amplify_auth_cognito/android/src/test/kotlin/com/amazonaws/amplify/amplify_auth_cognito/AmplifyAuthCognitoPluginTest.kt:33]: Android team has recommended against these "start" imports. I think there is a setting in Android Studio that will cause it to happen automatically that you may need to change.

**Review by haverchuck — APPROVED** [review]
LGTM

File: packages/amplify_auth_cognito/android/src/test/kotlin/com/amazonaws/amplify/amplify_auth_cognito/AmplifyAuthCognitoPluginTest.kt (hunk 1 above)
Comment (HuiSF) [reply packages/amplify_auth_cognito/android/src/test/kotlin/com/amazonaws/amplify/amplify_auth_cognito/AmplifyAuthCognitoPluginTest.kt:33]: +1
Can change the behavior here: Preferences -> Editor -> Code Style -> Kotlin

**Review by HuiSF — APPROVED** [review]

File: packages/amplify_auth_cognito/android/src/test/kotlin/com/amazonaws/amplify/amplify_auth_cognito/AmplifyAuthCognitoPluginTest.kt (hunk 1 above)
Comment (Jordan-Nelson) [reply packages/amplify_auth_cognito/android/src/test/kotlin/com/amazonaws/amplify/amplify_auth_cognito/AmplifyAuthCognitoPluginTest.kt]: Updated 👍

**Pull request merged**